### ISSUE TITLE
Placeholder: support using V1 tasks within V2 WorkflowTask

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-web#433]
(ref https://github.com/fractal-analytics-platform/fractal-server/issues/1328)

**Comment by tcompa** [comment]
This is not yet fully tested in the backend internal runner component, but the API should be mostly stable. Here is a first plan on how this could look in fractal-web.

In the V2 Workflow page, we should offer more options in the "add a task" modal.

1. By default, the user can pick a task from the V2 list (the current feature). V2 tasks can be compound, parallel or non-parallel (which then determines which parts of the arguments form should be available.
2. This choice, as we already had in V1, is split into common and users tasks.
3. By actively clicking on a switch, the modal should use the V1 task list (the one coming from `GET /api/v1/task/`) rather than the V2.
4. The split into common and users remains.
5. When adding a V1 task to a V2 workflow, the POST request body should include `"is_legacy_task": true`, see endpoint docs below.
6. When a V1 task is included in a V2 workflow, it is mostly treated as if a V2 task was included (note: it will always have `WorkflowTask.task_type="parallel"`). A couple of relevant differences:
    * The excluded arguments are the V1 ones (ref https://github.com/fractal-analytics-platform/fractal-web/issues/429).
    * It'd be useful to display somewhere in the task list that this task is legacy. I don't have any specific ideas.
    * As a first version, I think we should *not* offer version updates for V1 tasks which are run from a V2 workflow. I haven't checked yet whether this would be easy or complex.

![image](https://github.com/fractal-analytics-platform/fractal-web/assets/3862206/49c2e389-bb3a-493a-9cf4-c15fc6169977)


(side note: https://fractal-analytics-platform.github.io/fractal-server/openapi is now up-to-date)

**Comment by jluethi** [comment]
Great list. One addition:

> the modal should use the V1 task list (the one coming from GET /api/v1/task/) rather than the V2.

We have a flag that is set by admins on whether a V1 task is V2 compatible, right? We should only show the ones that have this flag set to True.

**Comment by tcompa** [comment]
> We have a flag that is set by admins on whether a V1 task is V2 compatible, right? We should only show the ones that have this flag set to True.

That's correct, thanks. We should filter the V1 task list and only include tasks with `is_v2_compatible=true`.

The V1 Task objects have a boolean property `is_v2_compatible`. This is always false, by default, and the only way to set it to true is that an admin should make a PATCH request to `/admin/v2/task-v1/{task_id}/` (this will be available in the V2 admin area - ref #438).

**Comment by tcompa** [comment]
Note that the relevant backend API was partly redefined in fractal-server 2.0.0a5, see
* https://github.com/fractal-analytics-platform/fractal-server/issues/1381
* https://github.com/fractal-analytics-platform/fractal-server/pull/1379/files

Now the relevant endpoint to get the list of V2-compatible V1 tasks is
```
GET /api/v2/task-legacy/?only_v2_compatible=true&args_schema=false
```
